pkg/api/routes.go

```diff
@@ -2204,6 +2211,79 @@ func (rh *RouteHandler) RevokeAPIKey(resp http.ResponseWriter, req *http.Request
 	resp.WriteHeader(http.StatusOK)
 }
 
+// ChangePassword godoc
+// @Summary Change user's password
+// @Description Change user's password. Invalidates all user's other active sessions.
+// @Router  /zot/auth/change_password [post]
+// @Accept  json
+// @Produce json
+// @Param old_password body string true "Old password"
+// @Param new_password body string true "New password"
+// @Success 200 {string} string "password changed"
+// @Failure 500 {string} string "internal server error"
+// @Failure 401 {string} string "unauthorized"
+// @Failure 400 {string} string "bad request"
+// @Failure 403 {string} string "old password is incorrect".
+func (rh *RouteHandler) ChangePassword(resp http.ResponseWriter, req *http.Request) {
+	// leave OPTIONS method
+	if req.Method == http.MethodOptions {
+		return
+	}
+
+	body, err := io.ReadAll(req.Body)
+	if err != nil {
+		rh.c.Log.Error().Msg("failed to read req body")
+		resp.WriteHeader(http.StatusInternalServerError)
+
+		return
+	}
+
+	var reqBody ChangePasswordRequest
+	if err := json.Unmarshal(body, &reqBody); err != nil {
+		rh.c.Log.Error().Msg("failed to unmarshal req body")
+		resp.WriteHeader(http.StatusBadRequest)
+
+		return
+	}
+
+	userAc, err := reqCtx.UserAcFromContext(req.Context())
+	if err != nil {
+		resp.WriteHeader(http.StatusNotFound)
+
+		return
+	}
+
+	username := userAc.GetUsername()
+	if username == "" {
+		resp.WriteHeader(http.StatusNotFound)
+	}
+
+	if err := rh.c.HtpasswdClient.ChangePassword(username, reqBody.OldPassword, reqBody.NewPassword); err != nil {
+		rh.c.Log.Error().Err(err).Str("identity", username).Msg("failed to change user password")
+		status := http.StatusInternalServerError
+
+		switch {
+		case errors.Is(err, zerr.ErrBadUser):
+			status = http.StatusNotFound
+		case errors.Is(err, zerr.ErrOldPasswordIsWrong):
+			status = http.StatusUnauthorized
+		case errors.Is(err, zerr.ErrPasswordIsEmpty):
+			status = http.StatusBadRequest
+		}
+
+		resp.WriteHeader(status)
+
+		return
+	}
+
+	resp.WriteHeader(http.StatusOK)
+}
```

Comment:
we should also clear user's sessions.

Reply:
get the session from the request.

set it's max age to 0 or -1 session.Options.MaxAge = -1
save it: session.Save()

you can look in authn.go to check how sessions are handled.
We want to clear it so that the user get's logout after changing he's password.